components/server/src/server.rs

```diff
@@ -608,6 +609,18 @@ where
             )),
         );
 
+        // Start the background check scheduler worker pool size periodically. This is
+        // only for the case that the resource control is enabled.
+        if self.core.config.resource_control.enabled {
+            let sched_pool = storage.get_scheduler().get_sched_pool().clone();
+            self.core.background_worker.spawn_interval_task(
+                DEFAULT_STORAGE_SCHED_POOL_SIZE_CHECK_INTERVAL,
+                move || {
+                    sched_pool.check_idle_pool();
```

Comment:
Pool size change event is already handled by the `scale_pool_size` function in dynamic config change handler, why still need this background task?

Reply:
Here I want to reduce the idle pool size,  such as by default, the priority queue size will scale in to 1.

Reply:
The pool size is actually means the maximum active thread numbers because all threads are already created, thus this optimize is totally useless here.

Reply:
👍  seems no need to do this. cc @Connor1996

Reply:
I thought the thread would exit with scale down ability introduced by https://github.com/tikv/yatp/pull/65, but seems it just park the thread.